### DISCUSSION TITLE
Autocomplete: backdrop isn't working as expected

### DISCUSSION THREAD

**Discussion by Holybasil** [heroui-inc/heroui#3705] [Feedback]
When I was using `Autocomplete`, I wanted the user to be able to focus on the entire component when the dropdown menu opened, so I used the `backdrop` attribute in `popoverProps`. However, that's where the problem arose.

Simply adding `backdrop: "opaque"` won't work unless you also set `isNonModal` to `false`. Because there is `isNonModal = true` in [use-popover.ts](https://github.com/nextui-org/nextui/blob/canary/packages/components/popover/src/use-popover.ts#L104) and it controls whether the backdrop content is rendered with the `backdrop` in [free-solo-popover.tsx](https://github.com/nextui-org/nextui/blob/canary/packages/components/popover/src/free-solo-popover.tsx#L144)
```js
popoverProps={{
    backdrop: "opaque",
    isNonModal: false,
}}
```

However, I can't see the backdrop content even if the two property values are set correctly. Because the backdrop content should be placed inside the outermost `div` of the popover, rather than becoming a sibling of this DOM element and being appended directly under the body.
![CleanShot 2024-09-03 at 16 48 41@2x](https://github.com/user-attachments/assets/31f6712f-2f1a-43b2-9bed-68c656123fd0)
which results in
![CleanShot 2024-09-03 at 16 25 55@2x](https://github.com/user-attachments/assets/af044700-ce23-40f5-9099-ea0b3006f8ba)

Additionally, even if the backdrop is positioned correctly, the component will look like this 
![CleanShot 2024-09-03 at 16 31 16@2x](https://github.com/user-attachments/assets/21a09a2c-cc7e-427e-90e7-49481178fb56)

instead of
![CleanShot 2024-09-03 at 16 32 22@2x](https://github.com/user-attachments/assets/58d0b111-94dc-4636-a08f-9bcd05ecc1fa)

From my side, I prefer the latter one.

Could someone take a look? If this happens to align with what your team wants, I can submit a pr.

**Comment by Holybasil** [comment]
here is the [codesandbox](https://codesandbox.io/p/devbox/nextui-autocomplete-backdrop-3c8fgp?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clzqdna2p00083j6uddql1oww%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clzqdna2o00033j6ua4mob1rv%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clzqdna2o00053j6umzy1rums%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clzqdna2o00073j6urc24e4qg%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clzqdna2o00033j6ua4mob1rv%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clzqdna2o00023j6urd6jho1z%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A58%252C%2522startColumn%2522%253A9%252C%2522endLineNumber%2522%253A64%252C%2522endColumn%2522%253A11%257D%255D%257D%252C%257B%2522id%2522%253A%2522clzqe6shr00143j6uip5r5o89%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A11%252C%2522startColumn%2522%253A16%252C%2522endLineNumber%2522%253A11%252C%2522endColumn%2522%253A16%257D%255D%252C%2522filepath%2522%253A%2522%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clzqdna2o00033j6ua4mob1rv%2522%252C%2522activeTabId%2522%253A%2522clzqdna2o00023j6urd6jho1z%2522%257D%252C%2522clzqdna2o00073j6urc24e4qg%2522%253A%257B%2522id%2522%253A%2522clzqdna2o00073j6urc24e4qg%2522%252C%2522activeTabId%2522%253A%2522cm0m5tai3000c3b6dipc6o7ek%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clzqdna2o00063j6u85oil456%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%257D%252C%257B%2522type%2522%253A%2522SETUP_TASKS%2522%252C%2522id%2522%253A%2522cm01o2z9d00093j6t6cx49coq%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522port%2522%253A5173%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522cm0m5sfxv000e3b6djkpirqil%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522port%2522%253A5173%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522cm0m5tai3000c3b6dipc6o7ek%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clzqdna2o00053j6umzy1rums%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clzqdna2o00043j6uwm3qa2mf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522id%2522%253A%2522clzqdna2o00053j6umzy1rums%2522%252C%2522activeTabId%2522%253A%2522clzqdna2o00043j6uwm3qa2mf%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by pablojsx** [comment]
Also backdrop from tailwind isn'tworking inside the navbar component

edit: I also noticed inset-0 isn't working properly either(inside the navbar)

**Comment by adnan-smlatic** [comment]
This helped me get the backdrop working, but then I couldn't close it, setting ```allowsCustomValue={true}``` on the autocomplete solved it. Just leaving it here if anyone has the same problem.

More at: https://github.com/heroui-inc/heroui/issues/1987#issuecomment-1817335810
